### ISSUE TITLE
Session ID keeps getting generated leading to 419

### ISSUE THREAD

**Issue by moay** [laravel/framework#38593]
- Laravel Version: 8.45+ (downgraded to 8.45, didn't test lower)
- PHP Version: 8.0.x (tested 8.0.9 and 8.0.10)
- Database Driver & Version: Doesn't matter, but MySQL running on MariaDB 10

### Description:
Currently we encounter a very (!) strange behaviour with the CSRF verification middleware. In an empty Laravel project (created using `composer create-project laravel/laravel`) we get session ID regeneration under very peculiar circumstances. Regardless of the actual session driver (tested file, cookie and database), cache driver (tested file and database), runtime environment (tested local server with `php artisan serve` on PHP 8.0.9 and 8.0.10 as well as a dockerized environment running on PHP8 as FPM over Apache), the CSRF token verification fails on basic form submissions. The token is set using a regular hidden field with name `_token`. The submission fails with a 419.

I tried really hard to pin this down. The main issue is that the session ID regenerates on every request. By debugging my way into the code I found the actual spot where the ID is regenerated and it's absolutely cumbersome. For the Laravel part it happens in the `\Illuminate\Foundation\Http\Middleware\VerifyCsrfToken` at line 73 (framework package v8.57). The `$this->isReading()` is actually triggering the session ID rewrite, that is the point where the session file is recreated and the old one is deleted. Digging deeper, it is actually happening in `vendor/symfony/http-foundation/Request.php` at line 1261.

What makes this even more strange is that it only happens when the `'POST' !== $this->method` at line 1261 is inside the if clause. As soon as I store the comparison outcome in a variable, then again it will fail when I put my variable as condition into the if clause. 

So, if I stop code execution at line 1260 (using an `exit()` or `die()`), the session file is still there. Line 1262 is never executed (because it's a post request), when stopping at line 1264, the session file changed. If I do the same comparison from line 1261 outside the if clause and store it as a variable, the session will not be regenerated.

I am pretty confident about my knowledge in PHP, but I fail to understand this. Am I right to assume some sort of bug in PHP itself? From the circumstances (different php runners, runtime environments, drivers for everything, and so on) it feels like maybe some file permission broke. But even when setting the entire project to 777, the issue persists. Are the some deep caches in some process cache that maybe can break this? (Switching PHP versions using homebrew didn't help, but I'm not sure whether homebrew actually kills anything when relinking the binaries)

Another strange one: I am able to solve the issue (at least temporarily) by checking out an earlier commit (this removes one unused controller). Bam, everything works again. This is no reliable solution as long as I don't understand the problem (nor the solution).

Does anyone have an idea what could have happened? I can't see how git could be the issue here...

### Steps To Reproduce:

Not easily reproducible, it just happened.

**Comment by driesvints** [comment]
Hi there, thanks for your report. Since we haven't gotten any other issues about this I'm going to close this one again. The best thing you can do is send in a PR with a failing test to prove the bug. If you can't then please try a support channel:

- [Laracasts Forums](https://laracasts.com/discuss)
- [Laravel.io Forums](https://laravel.io/forum)
- [StackOverflow](https://stackoverflow.com/questions/tagged/laravel)
- [Discord](https://discordapp.com/invite/KxwQuKb)
- [Larachat](https://larachat.co)
- [IRC](https://web.libera.chat/?nick=laravelnewbie&channels=#laravel)

Thanks.

**Comment by moay** [comment]
Ok, your answer forced me to think harder. Found the issue.

We had a custom user guard for local testing that would implement the `logout` method like this:

```php
public function logout() {
    Request::session()->invalidate();
}
```

While this way of logging out is of course not secure enough for real life, it works for our testing. By accident I figured that the session renew would only happen when at some place in the `routes/web.php` we restrict a route using the `auth` middleware. Doesn't even have to be the route using the CSRF token. With this function declaration and the `auth` middleware in the routes configuration, the session was actually getting regenerated on every request. BTW: The logout was never called. I still don't know how this broke it.

In case somebody finds this in the future...
